zokrates_core/src/proof_system/scheme/gm17.rs

```diff
@@ -0,0 +1,281 @@
+use proof_system::scheme::{Scheme, SolidityCompatibleScheme};
+use proof_system::solidity::{
+    SolidityAbi, SOLIDITY_G2_ADDITION_LIB, SOLIDITY_PAIRING_LIB, SOLIDITY_PAIRING_LIB_V2,
+};
+use proof_system::{G1Affine, G2Affine, G2AffineFq};
+use regex::Regex;
+use zokrates_field::{Bls12_377Field, Bls12_381Field, Bn128Field, Bw6_761Field, Field};
+
+pub trait NotBw6_761Field {}
+impl NotBw6_761Field for Bls12_377Field {}
+impl NotBw6_761Field for Bls12_381Field {}
+impl NotBw6_761Field for Bn128Field {}
+
+pub struct GM17;
+
+#[derive(Serialize, Deserialize)]
+pub struct ProofPoints<G1, G2> {
+    pub a: G1,
+    pub b: G2,
+    pub c: G1,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct VerificationKey<G1, G2> {
+    pub h: G2,
+    pub g_alpha: G1,
+    pub h_beta: G2,
+    pub g_gamma: G1,
+    pub h_gamma: G2,
+    pub query: Vec<G1>,
+}
+
+impl<T: Field + NotBw6_761Field> Scheme<T> for GM17 {
+    type VerificationKey = VerificationKey<G1Affine, G2Affine>;
+    type ProofPoints = ProofPoints<G1Affine, G2Affine>;
+}
+
+impl Scheme<Bw6_761Field> for GM17 {
+    type VerificationKey = VerificationKey<G1Affine, G2AffineFq>;
+    type ProofPoints = ProofPoints<G1Affine, G2AffineFq>;
+}
+
+impl<T: Field + NotBw6_761Field> SolidityCompatibleScheme<T> for GM17 {
```

Comment:
To work with solidity, we must be using Bn128Field. This allows more than that (for example BLS12-377)